software/py/blood_graph_key.py

```diff
@@ -0,0 +1,48 @@
+import sys
+import argparse
+from blood_graph import BloodGraph
+from PIL import Image, ImageDraw, ImageFont
+from itertools import chain
+
+WIDTH  = 256
+HEIGHT = 256
+DEFAULT_MODE = "detailed"
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Depicts a visual key for bloodgraph")
+    parser.add_argument("--mode", default=DEFAULT_MODE, type=str,
+                        help="Type of bloodgraph - abstract / detailed")
+    args = parser.parse_args()
+
+    if (not (args.mode == "abstract" or args.mode == "detailed")):
```

Comment:
Fun fact, you can write custom parsers so that you don't need to do this. 

https://docs.python.org/3.6/library/argparse.html#action

Reply:
Yeah I used argparse in the blood_graph.py, was too lazy to do the same for blood_graph_key.py, it's merged into blood_graph.py now (thanks to Max) so we're good.